cubito/main.cpp

```diff
@@ -208,73 +216,81 @@ int main() {
 
     rubick_cube.cubitos[1] = std::make_shared<Cubito>(cubito_program,
-      glm::vec3(-0.5f, 0.5f, 0.5f), 1, input_colors[12], input_colors[11],
-      input_colors[6]);
+                                                      glm::vec3(-0.5f, 0.5f, 0.5f), 1, input_colors[12], input_colors[11],
+                                                      input_colors[6]);
     rubick_cube.cubitos[4] = std::make_shared<Cubito>(cubito_program,
-      glm::vec3(0.0f, 0.5f, 0.5f), 4, input_colors[13], input_colors[7]);
+                                                      glm::vec3(0.0f, 0.5f, 0.5f), 4, input_colors[13], input_colors[7]);
     rubick_cube.cubitos[7] = std::make_shared<Cubito>(cubito_program,
-      glm::vec3(0.5f, 0.5f, 0.5f), 7, input_colors[14], input_colors[15],
-      input_colors[8]);
+                                                      glm::vec3(0.5f, 0.5f, 0.5f), 7, input_colors[14], input_colors[15],
+                                                      input_colors[8]);
     rubick_cube.cubitos[10] = std::make_shared<Cubito>(cubito_program,
-      glm::vec3(-0.5f, 0.0f, 0.5f), 10, input_colors[24], input_colors[23]);
+                                                       glm::vec3(-0.5f, 0.0f, 0.5f), 10, input_colors[24], input_colors[23]);
     rubick_cube.cubitos[13] = std::make_shared<Cubito>(cubito_program,
-      glm::vec3(0.0f, 0.0f, 0.5f), 13, input_colors[25]);
+                                                       glm::vec3(0.0f, 0.0f, 0.5f), 13, input_colors[25]);
     rubick_cube.cubitos[16] = std::make_shared<Cubito>(cubito_program,
-      glm::vec3(0.5f, 0.0f, 0.5f), 16, input_colors[26], input_colors[27]);
+                                                       glm::vec3(0.5f, 0.0f, 0.5f), 16, input_colors[26], input_colors[27]);
     rubick_cube.cubitos[19] = std::make_shared<Cubito>(cubito_program,
-      glm::vec3(-0.5f, -0.5f, 0.5f), 19, input_colors[36], input_colors[35],
-      input_colors[45]);
+                                                       glm::vec3(-0.5f, -0.5f, 0.5f), 19, input_colors[36], input_colors[35],
+                                                       input_colors[45]);
     rubick_cube.cubitos[22] = std::make_shared<Cubito>(cubito_program,
-      glm::vec3(0.0f, -0.5f, 0.5f), 22, input_colors[37], input_colors[46]);
+                                                       glm::vec3(0.0f, -0.5f, 0.5f), 22, input_colors[37], input_colors[46]);
     rubick_cube.cubitos[25] = std::make_shared<Cubito>(cubito_program,
-      glm::vec3(0.5f, -0.5f, 0.5f), 25, input_colors[38], input_colors[39],
-      input_colors[47]);
+                                                       glm::vec3(0.5f, -0.5f, 0.5f), 25, input_colors[38], input_colors[39],
+                                                       input_colors[47]);
   }
   
   PrintCommands();
 
+  StartParser();
+
+  std::string str_scramble = (argc == 2 ? argv[1] : GenScramble(15));

```

Comment:
Also, it would be good to have some safety-check that the sequence is ok because it has broken my machine many times and I have to completely shut it down. If it is not ok, just print on console and generate a random sequence.

Reply:
Done